apps/api/src/app/notifications/usecases/get-activity-feed/get-activity-feed.usecase.ts

```diff
@@ -46,6 +67,72 @@ export class GetActivityFeed {
     };
   }
 
+  private async validateRetentionLimitForTier(organizationId: string, after?: string, before?: string) {
+    const organization = await this.organizationRepository.findById(organizationId);
+
+    if (!organization) {
+      throw new HttpException('Organization not found', HttpStatus.INTERNAL_SERVER_ERROR);
+    }
+
+    const maxRetentionMs = this.getMaxRetentionPeriodByOrganization(organization);
+
+    const earliestAllowedDate = new Date(Date.now() - maxRetentionMs);
+
+    // If no after date is provided, default to 24 hours ago
+    const effectiveAfterDate = after ? new Date(after) : DEFAULT_AFTER_DATE;
+    const effectiveBeforeDate = before ? new Date(before) : new Date();
```

Comment:
Since after and before are optional fields in DTO, user would fetch all notifications for his organization by default.

Starting by the smallest possible range is safer and puts less load on DB, while user is still allowed to increase the range.